src/EnergyPlus/UnitarySystem.hh

```diff
@@ -0,0 +1,684 @@
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#ifndef ENERGYPLUS_UNITARYSYSTEM_HH
+#define ENERGYPLUS_UNITARYSYSTEM_HH
+
+#include <string>
+#include <vector>
+
+namespace EnergyPlus {
+
+namespace UnitarySystems {
+
+    //extern int numDesignSpecMultiSpeedHP;
+    extern int numUnitarySystems;
+    //bool getInputFlag;
+    extern bool economizerFlag; // holds air loop economizer status
+    extern bool SuppHeatingCoilFlag; // set to TRUE when simulating supplemental heating coil
+    
+    // why are these external?
+    // Last mode of operation
+    extern int const CoolingMode; // last compressor operating mode was in cooling
+    extern int const HeatingMode; // last compressor operating mode was in heating
+
+    // Compressor operation
+    extern int const On;  // normal compressor operation
+    extern int const Off; // signal DXCoil that compressor shouldn't run
+
+    // Coil type for SimWater and SimSteamCoil
+    extern int const CoolingCoil;
+    extern int const HeatingCoil;
+    extern int const SuppHeatCoil;
+
+    // Supply Air Sizing Option
+    extern int const None;
+    extern int const SupplyAirFlowRate;
+    extern int const FlowPerFloorArea;
+    extern int const FractionOfAutoSizedCoolingValue;
+    extern int const FractionOfAutoSizedHeatingValue;
+    extern int const FlowPerCoolingCapacity;
+    extern int const FlowPerHeatingCapacity;
+
+    class DesignSpecMSHP
+    {
+        friend class UnitarySys;
+
+    public:
+        DesignSpecMSHP(); // constructor
+        ~DesignSpecMSHP() // destructor
+        {
+        }
+
+        std::string name;
+        static DesignSpecMSHP *factory(int object_type_of_num, std::string const objectName);
+
+    private:
+        int designSpecMSHPType_Num;
+        Real64 noLoadAirFlowRateRatio;
+        int numOfSpeedHeating;
+        int numOfSpeedCooling;
+        std::vector<Real64> coolingVolFlowRatio; // The ratio of flow to max for this speed
+        std::vector<Real64> heatingVolFlowRatio; // The ratio of flow to max for this speed
+        // std::vector<Real64> coolVolumeFlowRate;
+        //std::vector<Real64> coolMassFlowRate;
+        std::vector<Real64> MSCoolingSpeedRatio;
+        //std::vector<Real64> heatVolumeFlowRate;
+        //std::vector<Real64> heatMassFlowRate;
+        std::vector<Real64> MSHeatingSpeedRatio;
+        bool singleModeFlag;
+
+        static void getDesignSpecMSHP();
+        static void getDesignSpecMSHPdata(bool errorsFound);
+    };
+
+    class UnitarySys
+    {
+
+        // bool myOneTimeFlag;
+        // bool getInputOnceFlag;
+        int fanSpeedRatio;
+
+        enum controlTypeEnum : int
+        {
+            controlTypeNone,
```

Comment:
First, I like the use of `enum`s here, but use an `enum class` instead and drop the `controlType` prefix from the members and `enum` from the name. Also, capitalize (`ControlType`, `None`, `Load`, etc.).

```
enum class ControlType : int
{
    None,
    TypeLoad,
    Setpoint,
    CCMASHRAE
};
```

Reply:
OK, this one should be easy to change. I don't know the difference between enum and enum class.